### PR TITLE
Github actions

### PR DESCRIPTION
**Pull request by kcgroulx**
## Pull Request Description

This PR removes the broken build actions for MCB and adds a new one for building the firmware projects. It uses a premade Github action that can be found here: https://github.com/xanderhendriks/action-build-stm32cubeide.
You can see the build in the Checks tab of a PR.

## Monday Link

https://ubcsolar26.monday.com/boards/7524367653/pulses/7714173981

## Effected Components

- [ ] AMB
- [ ] BMS
- [ ] DID
- [ ] ECU
- [ ] MCB
- [ ] MDI
- [X] TEL
- [X] CI/CD

## Testing

- [ ] Vehicle testing
- [ ] Bench top testing
- [ ] Unit testing
- [X] Other
- [ ] N/A

Pushed bad TEL code and verify that the Github action had an error.

## Sanity check

- CAN ID table updated
- IOC updated and commited
- gitignore updated and commited
- [X] Steps confirmed

## Sources

N/A

### CODE REVIEW

**Review by AarjavJain101 — APPROVED** [review]
heat

File: libraries/CAN_comms/Inc/CAN_comms.h [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 
 #include <stdint.h>
-#include "cmsis_os2.h"
+#include "cmsis_os.h"
```
Comment (makac1896) [inline libraries/CAN_comms/Inc/CAN_comms.h:14]: Why are we changing this in this PR?

File: .github/workflows/firmware.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+jobs:
+  build_components:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v3.1.0
+      - name: Build TEL
+        uses: xanderhendriks/action-build-stm32cubeide@v13.0
+        with:
```
Comment (makac1896) [inline .github/workflows/firmware.yml:13]: question: will we be extending this functionality to all other boards in the future or just tel for now?

File: libraries/CAN_comms/Inc/CAN_comms.h (hunk 1 above)
Comment (makac1896) [reply libraries/CAN_comms/Inc/CAN_comms.h:14]: Never mind, just saw from your commit history that this change allows CAN_comms to build. Lgtm! Thanks Kyle!

**Review by makac1896 — APPROVED** [review]
approved

File: .github/workflows/firmware.yml (hunk 2 above)
Comment (kcgroulx) [reply .github/workflows/firmware.yml:13]: Yes, the plan is that the new boards (DRD, STR, etc...) will be added to this workflow. Old boards (MCB, DID) will not be added.

**Pull request merged**